### ISSUE TITLE
Kernel crashes and needed to restart after using KernelPCA

### ISSUE THREAD

**Issue by Dedsd** [microsoft/vscode-jupyter#7802]
OS: Ubuntu 20.04
Vscode: 1.60.2
Extension: 2021.8.1236758218

After executing some code of KernelPCA with sklearn library I get this kernel crash that needed to restart the kernel(if I didin't restart keep the error message while executing any cell). After restarting it continues to get the crash when executing the KernelPCA code. So I can't execute the code without crashing.

Error message:
```
Error: Session cannot generate requests
Error: Session cannot generate requests
at w.executeCodeCell (/home/deds/.vscode/extensions/ms-toolsai.jupyter-2021.8.1236758218/out/client/extension.js:90:327199)
at w.execute (/home/deds/.vscode/extensions/ms-toolsai.jupyter-2021.8.1236758218/out/client/extension.js:90:326520)
at w.start (/home/deds/.vscode/extensions/ms-toolsai.jupyter-2021.8.1236758218/out/client/extension.js:90:322336)
at processTicksAndRejections (internal/process/task_queues.js:93:5)
at async t.CellExecutionQueue.executeQueuedCells (/home/deds/.vscode/extensions/ms-toolsai.jupyter-2021.8.1236758218/out/client/extension.js:90:336863)
at async t.CellExecutionQueue.start (/home/deds/.vscode/extensions/ms-toolsai.jupyter-2021.8.1236758218/out/client/extension.js:90:336403)
```

**Comment by DavidKutu** [comment]
Thanks for filing @Dedsd.

Could you share what is the code that is crashing?

**Comment by Dedsd** [comment]
This code crashes the kernel:
```Python
kernelpca = KernelPCA(n_components=8, kernel='rbf')
x_census_train_kernelpca = kernelpca.fit_transform(x_census_train)
x_census_test_kernelpca = kernelpca.transform(x_census_test)
```

**Comment by DonJayamanne** [comment]
>  fix this error printed with sklearn 1.0.1 it will still give the requests problem, because I think it is the extension, I never had this error using the default jupyter..

Please could you confirm this works in classic Jupyter. (outside of VS Code)

**Comment by Dedsd** [comment]
Yes. It works without crashing in classic jupyter.

**Comment by DonJayamanne** [comment]
> Yes. It works fine and without crashing in classic jupyter.

Please could you do the following:
* Inside Jupyter, run a cell with the following code
```
import sys
print(sys.executable)
```
* Where are you starting Jupyter from
	* What python environment is Jupyter installed it
	*  How are you launching jupyter (are you using `python -m jupyter notebook` or just `jupyter notebook` or the like
	*

**Comment by Dedsd** [comment]
Python 3.8.10 in `/usr/bin/python3`
I was using `jupyter notebook`

**Comment by DonJayamanne** [comment]
@Dedsd 

Please could you try this in VS Code insiders.
* Install VSCode insiders from here https://code.visualstudio.com/insiders/
* Install Jupyter extesnsion (in VS Code Insiders you'll automatically get the latest dev version of Jupyter extension built daily from our main branch)
* Open settings & go into `Jupyter->Logging`
* Change the value to `verbose`
* Reload VS Code,
* Attempt to repro the issue & then copy the output from the `Jupyter` output panel.you install VS Code insiders

Please copy all of the output.
Thanks

**Comment by Dedsd** [comment]
In vscode 1.63 strangely the issue is gone. Seems to be fixed.

**Comment by Dedsd** [comment]
I don't know if it was a problem in vscode 1.60..

**Comment by DonJayamanne** [comment]
@Dedsd We've released new versions of the extension since and the new versions only work in later versions of VS Code.
Hence its possible the fix isn't available in 1.60.

Thanks for getting back to us. Closing this issue for now.
